File: dotnet/Samples/LanguageDetection/Text/TextLanguageDetectionService.cs
using LanguageWeaver.Sdk.Common;
using LanguageWeaver.Sdk.Configurations;
using LanguageWeaver.Sdk.Constants;
using LanguageWeaver.Sdk.LanguageDetection.Common.Request;

namespace LanguageWeaver.Sdk.Samples.LanguageDetection.Text;

public class LanguageDetectionService
{
    public static void Main()
    {
        using var lwClient = new SdkFactory().GetLanguageWeaverClient(new ClientConfiguration());
        var textLanguageDetectionRequest = new TextLanguageDetectionRequest()
        {
            Input = "A mountain is generally steeper than a hill",
            InputFormat = Format.Plain
        };

        var languageDetectionResult = lwClient.DetectLanguage(textLanguageDetectionRequest);

        languageDetectionResult.Languages
            .ForEach(language => Console.WriteLine(language.Name));
    }
}

File: javascript/backend/samples/language-detection/text/TextLanguageDetectionService.js
const SdkFactory = require("@language-weaver/lw-sdk-js").default;
const {
    TextLanguageDetectionRequest,
    Format,
    ClientConfiguration,
} = require("@language-weaver/lw-sdk-js");
const detectTextLanguage = async () => {
    try{
        const lwClient = await SdkFactory.getLanguageWeaverClient(new ClientConfiguration());
        const textLanguageDetectionRequest = new TextLanguageDetectionRequest();
        textLanguageDetectionRequest.input = "A mountain is generally steeper than a hill";
        textLanguageDetectionRequest.inputFormat = Format.PLAIN;

        const textLanguageDetectionResult = await lwClient.detectTextLanguage(textLanguageDetectionRequest);
        console.log(textLanguageDetectionResult);
    } catch (e) {
        console.log(e);
    }
}

detectTextLanguage();